### PR TITLE
feat(#major); abracadabra; subgraph optimizations

### PR DESCRIPTION
**Pull request by dhruv-chauhan**
- Huge arrays in `abracadabra-money-avalanche` deployment causing issues for other subgraphs in the shard.
- As a [suggested storage optimization](https://docs.google.com/document/d/1xJif1OtgQl0hEEjtK1D71b7eGII_J0CT4Ibqj3RzfR8/edit), we looked to convert large arrays (length>50) as `@derivedFrom` mappings, but on querying it looks like the larger arrays already follow this convention, others do not need to due to very small length. [Reference query](https://api.thegraph.com/subgraphs/id/QmUjj4BbyeAEeBxFtgZeG7e4akyU29Y7fwe4utPN8qYNWX/graphql?query=%7B%0A++lendingProtocols%7B%0A++++mintedTokens%7Bid%7D%0A++++mintedTokenSupplies%0A++++marketIDList%0A++++%0A++++%23%23%23+derived%0A++++%23+dailyUsageMetrics%7Bid%7D%0A++++%23+hourlyUsageMetrics%7Bid%7D%0A++++%23+financialMetrics%7Bid%7D%0A++++%23+markets%7Bid%7D%0A++%7D%0A++financialsDailySnapshots%28first%3A1%2C+orderBy%3AblockNumber%2C+orderDirection%3Adesc%29%7B%0A++++mintedTokenSupplies%0A++%7D%0A++markets%7B%0A++++rewardTokens%7Bid%7D%0A++++rates%7Bid%7D%0A++++rewardTokenEmissionsAmount%0A++++rewardTokenEmissionsUSD%0A++++%0A++++%23+%23%23%23+derived%0A++++%23+positions%7Bid%7D%0A++++%23+dailySnapshots%7Bid%7D%0A++++%23+hourlySnapshots%7Bid%7D%0A++++%23+deposits%7Bid%7D%0A++++%23+withdraws%7Bid%7D%0A++++%23+borrows%7Bid%7D%0A++++%23+repays%7Bid%7D%0A++++%23+liquidates%7Bid%7D%0A++%7D%0A++marketDailySnapshots%28first%3A1%2C+orderBy%3AblockNumber%2C+orderDirection%3Adesc%29%7B%0A++++rates%7Bid%7D%0A++++rewardTokenEmissionsAmount%0A++++rewardTokenEmissionsUSD%0A++%7D%0A++marketHourlySnapshots%28first%3A1%2C+orderBy%3AblockNumber%2C+orderDirection%3Adesc%29%7B%0A++++rates%7Bid%7D%0A++++rewardTokenEmissionsAmount%0A++++rewardTokenEmissionsUSD%0A++%7D%0A++%23+accounts%7B%0A++%23+%09%23%23%23+derived%0A++%23+++positions%7Bid%7D%0A++%23+++deposits%7Bid%7D%0A++%23+++withdraws%7Bid%7D%0A++%23+++borrows%7Bid%7D%0A++%23+++repays%7Bid%7D%0A++%23+++liquidates%7Bid%7D%0A++%23+++liquidations%7Bid%7D%0A++%23+%7D%0A++%23+positions%7B%0A++%23+++%23%23%23+derived%0A++%23+++deposits%7Bid%7D%0A++%23+++withdraws%7Bid%7D%0A++%23+++borrows%7Bid%7D%0A++%23+++repays%7Bid%7D%0A++%23+++liquidations%7Bid%7D%0A++%23+++snapshots%7Bid%7D%0A++%23+%7D%0A%7D); [schema](https://github.com/messari/subgraphs/blob/master/subgraphs/abracadabra/schema.graphql).
- Since most of these large arrays are snapshots and events, we could further make these entities immutable.
- ~~Changed ID fields from strings to Bytes.~~
- Test deployment: https://okgraph.xyz/?q=dhruv-chauhan%2Fabracadabra-money-avalanche

### CODE REVIEW

File: subgraphs/abracadabra/schema.graphql [hunk 1]
```diff
@@ -37,7 +37,7 @@ enum ProtocolType {
 
 type Token @entity @regularPolling {
   " Smart contract address of the token "
-  id: ID!
+  id: Bytes!
```
Comment (melotik) [inline subgraphs/abracadabra/schema.graphql:40]: Just to confirm, the casing of IDs or IDs themselves do not change with this update to Bytes?

**Review by melotik — COMMENTED** [review]
code changes lgtm and the schema looks good. Just want to confirm the comment I made. If the IDs change at all we have assumptions downstream that need to be coordinated

File: subgraphs/abracadabra/schema.graphql (hunk 1 above)
Comment (dhruv-chauhan) [reply subgraphs/abracadabra/schema.graphql:40]: @melotik ID values will change.
Eg, (interestRates) `"BORROWER-STABLE-0x13eca380e816edffc356b75b2fcd84b5d3c464d9-18900"` -> `0x00000000424f52524f5745522d535441424c452d13eca380e816edffc356b75b2fcd84b5d3c464d92d3138393030`

Comment (melotik) [reply subgraphs/abracadabra/schema.graphql:40]: My sense is that the downstream affects are too costly, so we should just leave it as is. But looking for confirmation from @jaksonkallio

Comment (dhruv-chauhan) [reply subgraphs/abracadabra/schema.graphql]: Reverted

**Review by melotik — APPROVED** [review]

**Pull request merged**